libs/dal/src/repositories/job/job.schema.ts

```diff
@@ -143,5 +134,80 @@ jobSchema.virtual('environment', {
   justOne: true,
 });
 
+/*
+ * This index was initially created to optimize:
+ * apps/api/src/app/events/usecases/add-job/add-delay-job.usecase.ts
+ */
+jobSchema.index({
+  transactionId: 1,
+  _subscriberId: 1,
```

Comment:
Because of the cardinality of those 2 fields together I feel like we don't need all the other properties here of `_templateId` and `_environmentId` unless this could become a covered query. So having `transactionId, _subscriberId, status` should be enough in my opinion